lib/buffer.js

```diff
@@ -379,6 +378,16 @@ Buffer.prototype.toString = function(encoding, start, end) {
 };
 
 
+Buffer.prototype.toString = function() {
+  const length = this.length | 0;
+  if (length === 0)
+    return '';
+  if (arguments.length === 0)
+    return this.utf8Slice(0, length);
+  return slowToString.apply(this, arguments);
```

Comment:
See [this comment](https://github.com/nodejs/io.js/pull/2027#discussion_r32898233).  The initial version called `slowToString(that, arguments[0], ...)` but when I ran more benchmarks, it turned out that `.apply()` is faster by about 25-30% once the optimizing compiler kicks in.